### PR TITLE
PR to merge forms into develop

### PR DESCRIPTION
**Pull request by UmairAhmad125**
In this project, we build a form that takes the username, email, and password of the user, validates the data, and saves in the database.
User can also update user data.

### CODE REVIEW

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        <%= pluralize(@user.errors.count, "error") %> prohibited
+        this article from being saved:
+      </h2>
+      <ul>
+        <% @user.errors.full_messages.each do |msg| %>
+          <li><%= msg %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (lekegitrepo) [inline app/views/users/edit.html.erb]: - **_There should be NO application logic in the views instead create a helper, logic can be encapsulated in View helpers_**

- **_There should be commented out `form` with:_**
* [x] **_uses `users_path` helper for action_**
* [x] **_uses `post` for method_**
* [x] **_uses `"UTF-8"` for ccept-charset_**
* [x] **_uses `user[<field_name>]` for input's name_**
* [x] **_uses form_authenticity_token_**

- **_There should be commented out `form_tag` that uses `label_tag`, `text_field_tag`, `email_field_tag`, `password_field_tag`_**

### Optional
* [ ] **_partial created for user form and reused in edit and new view_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Status: Request changes :recycle:
#### Hi @UmairAhmad125 and @talhawaqar :clap: :clap:
#### Good job so far. However, you need to implement a few requirement before this project can be approved.

#### Check my comments below :point_down:

#### _Reviewed by Temitayo Adeleke_.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to new_user_path
```
Comment (uryelah) [inline app/controllers/users_controller.rb]: - [x] You should be redirecting to `redirect_to edit_user_path` instead.

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+
+<%# <form action="/users" method="post" accept-charset="UTF-8">
```
Comment (uryelah) [inline app/views/users/new.html.erb]: - [x] The action used here should be the helper `users_path`

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+<input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>
+  <%# <label for="username">Username:</label>
+  <input type="text" id="username" name="username"><br><br>
```
Comment (uryelah) [inline app/views/users/new.html.erb]: - [x] The names of each input should follow the pattern `user[<field_name>]`.
- [x] Make sure this commented out form is actually working.
You should have followed the work flow Odin project gave you, 
first make sure the form using `<form>` tag is working.
then comment it out
then make the form_tag form
make sure it's working
and the comment it out.

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  <label for="email">Email:</label>
+  <input type="email" id="email" name="email"><br><br>
+  <label for="password">Pasword:</label>
+  <input type="password" id="password" name="password"><br><br>
+  <input type="submit" value="Submit">
+</form> %>
+
+<%# <%= form_tag("/users", method: "post") do %>
```
Comment (uryelah) [inline app/views/users/new.html.erb]: - [x] Use `users_path` for the form_tag only parameter. Adding the method won't be needed then.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Umair Ahmad & Talha Waqar**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with some of the fixes, but there are still some requirements from Odin project instructions missing in your project. Make sure to read the comments under this post carefully and make the needed changes.

# Required Changes 🏗️ 🐛 💡
(Check below)

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by Ispirett — APPROVED** [review]
## Status Approved 👍 

Good job folks incorporating all the changes requested, you have done well.  🥇

**Pull request merged**